engine/src/test/java/org/hibernate/search/testsupport/TestConstants.java

```diff
@@ -89,4 +85,11 @@ public static String getIndexDirectory(Class<?> testClass) {
 	public static boolean arePerformanceTestsEnabled() {
 		return Boolean.getBoolean( "org.hibernate.search.enable_performance_tests" );
 	}
+
+	private static class CallerProvider extends SecurityManager {
+
+		public Class<?> getCallerClass() {
+			return getClassContext()[2];
```

Comment:
But `CallerProvider` is stateless, it does not point to any directory. It is just used to obtain the method of the calling test which wants to create an index directory (so its name can be added to the dir name). It's not related to the target dir.